pkg/operator/ceph/file/subvolumegroup/controller.go

```diff
@@ -201,10 +212,24 @@ func (r *ReconcileCephFilesystemSubVolumeGroup) reconcile(request reconcile.Requ
 	// DELETE: the CR was deleted
 	if !cephFilesystemSubVolumeGroup.GetDeletionTimestamp().IsZero() {
 		logger.Debugf("deleting subvolume group %q", namespacedName)
+
+		cephFsSvgList := &cephv1.CephFilesystemSubVolumeGroupList{}
+		// List cephFilesystemSubvolumeGroup CR based on filesystem and spec.name
+		matchingKey := fmt.Sprintf("%s/%s", getSubvolumeGroupName(cephFilesystemSubVolumeGroup), cephFilesystemSubVolumeGroup.Spec.FilesystemName)
+		err = r.client.List(r.opManagerContext, cephFsSvgList, &client.MatchingFields{cephSVGFileSystemNameIndex: matchingKey})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return reconcile.Result{}, errors.Wrap(err, "failed to list cephFilesystemSubvolumeGroup")
+		}
```

Comment:
```suggestion
		if err = r.client.List(r.opManagerContext, cephFsSvgList, &client.MatchingFields{cephSVGFileSystemNameIndex: matchingKey}); client.IgnoreNotFound(err) != nil {
			return reconcile.Result{}, errors.Wrap(err, "failed to list cephFilesystemSubvolumeGroup")
		}
```

Reply:
We have two checks`if err != nil && !kerrors.IsNotFound(err) {` so added like this only.

Reply:
Are we going to get notfound error when we are doing list operation?

Reply:
well if there is no svg with matchingFields, then we'll get the `notFound` err right

Reply:
@rewantsoni I update the code but still spliting the check in 2 line for better reading

Reply:
@subhamkrai Madhu is right, we should not get a not found error for the list operation, the list will be empty